### ISSUE TITLE
undefined reference to `QKeychain:: ...

### ISSUE THREAD

**Issue by algo99** [psi-im/psi#511]
Trying to build from master on Fedora-32 (Fedora-31 gives the same error), I get _undefined reference_ error on link step, although the configuration step looks Ok:

```
$ export QTDIR= && ./configure --prefix=/home/dist --enable-webkit --release --no-separate-debug-info --qtselect=5
Configuring Psi ...
Verifying Qt build environment ... ok
Checking for Qt >= 5.0.0 ... 5.13.2
Checking for LibIDN ... yes
Checking for QCA >= 2.0 ... yes
Checking for zlib ... yes
Checking for QJDns ... Disabled for Qt5 and above
Checking for QtDbus ... yes
Checking for Qt Keychain ... yes
Checking for QtMultimedia ... yes
Checking for QtConcurrent ... yes
Checking for QtSql ... yes
Checking for QtWidgets ... yes
Checking for QtNetwork ... yes
Checking for QtXml ... yes
Checking for QtWebKit ... QtWebEngine
Checking for HTTP Parser ...HTTP Parser includes not found. Use bundled
 bundled
Checking for the XScreenSaver extension ... no
Checking for spellcheck engine ... hunspell
Checking for Psi Plugin support ... yes

Good, your configure finished.  Now run /usr/bin/gmake.
```

<details>
  <summary>Details of linking</summary>

```sh

g++ -Wl,-O1 -Wl,-rpath-link,/usr/lib64 -o /home/albert/Downloads/git/psi/psi .obj/iris_discoinfoquerier.o .obj/privacylistitem.o .obj/privacylist.o .obj/privacylistmodel.o .obj/privacylistblockedmodel.o .obj/psiprivacymanager.o .obj/privacydlg.o .obj/privacyruledlg.o .obj/tabdlg.o .obj/tabmanager.o .obj/tabbablewidget.o .obj/CertificateDisplayDialog.o .obj/CertificateErrorDialog.o .obj/CertificateHelpers.o .obj/contactmanagerdlg.o .obj/contactmanagermodel.o .obj/contactmanagerview.o .obj/idle.o .obj/idle_x11.o .obj/systemwatch.o .obj/systemwatch_unix.o .obj/zip.o .obj/unzip.o .obj/ioapi.o .obj/globalshortcutmanager.o .obj/globalshortcutmanager_x11.o .obj/spellchecker.o .obj/spellhighlighter.o .obj/hunspellchecker.o .obj/atomicxmlfile.o .obj/simplecli.o .obj/priorityvalidator.o .obj/iodeviceopener.o .obj/languagemanager.o .obj/iconset.o .obj/anim.o .obj/optionstree.o .obj/varianttree.o .obj/optionstreereader.o .obj/optionstreewriter.o .obj/optionstreemodel.o .obj/advwidget.o .obj/grepshortcutkeydialog.o .obj/psimedia.o .obj/jinglertptasks.o .obj/jinglertp.o .obj/avcall.o .obj/calldlg.o .obj/mediadevicewatcher.o .obj/pollingtunecontroller.o .obj/filetunecontroller.o .obj/tunecontrollermanager.o .obj/psifileplugin.o .obj/mpristunecontroller.o .obj/mprisplugin.o .obj/AutoUpdater.o .obj/busywidget.o .obj/fancylabel.o .obj/iconselect.o .obj/iconwidget.o .obj/psitextview.o .obj/urllabel.o .obj/urlobject.o .obj/iconaction.o .obj/fancypopup.o .obj/psirichtext.o .obj/psitooltip.o .obj/psitiplabel.o .obj/psitabwidget.o .obj/psitabbar.o .obj/actionlineedit.o .obj/tabbar.o .obj/typeaheadfind.o .obj/pixmapratiolabel.o .obj/debug.o .obj/varlist.o .obj/jidutil.o .obj/showtextdlg.o .obj/psi_profiles.o .obj/activeprofiles.o .obj/profiledlg.o .obj/aboutdlg.o .obj/desktoputil.o .obj/fileutil.o .obj/textutil.o .obj/pixmaputil.o .obj/accountscombobox.o .obj/psievent.o .obj/xmlconsole.o .obj/psiiconset.o .obj/psithememanager.o .obj/psithememodel.o .obj/psithemeprovider.o .obj/theme.o .obj/theme_p.o .obj/applicationinfo.o .obj/pgptransaction.o .obj/userlist.o .obj/mainwin.o .obj/mainwin_p.o .obj/psitrayicon.o .obj/rtparse.o .obj/systeminfo.o .obj/common.o .obj/proxy.o .obj/miniclient.o .obj/accountmanagedlg.o .obj/accountadddlg.o .obj/accountregdlg.o .obj/accountmodifydlg.o .obj/changepwdlg.o .obj/htmltextcontroller.o .obj/msgmle.o .obj/chatviewcommon.o .obj/messageview.o .obj/statusdlg.o .obj/statuscombobox.o .obj/eventdlg.o .obj/activecontactsmenu.o .obj/chatdlg.o .obj/psichatdlg.o .obj/chatsplitter.o .obj/chateditproxy.o .obj/adduserdlg.o .obj/mcmdmanager.o .obj/mcmdsimplesite.o .obj/infodlg.o .obj/translationmanager.o .obj/eventdb.o .obj/edbflatfile.o .obj/edbsqlite.o .obj/historydlg.o .obj/historyimp.o .obj/historycontactlistmodel.o .obj/searchdlg.o .obj/registrationdlg.o .obj/psitoolbar.o .obj/passphrasedlg.o .obj/vcardfactory.o .obj/discodlg.o .obj/alerticon.o .obj/alertable.o .obj/psipopup.o .obj/popupmanager.o .obj/psipopupinterface.o .obj/psiapplication.o .obj/filecache.o .obj/avatars.o .obj/actionlist.o .obj/psiactionlist.o .obj/xdata_widget.o .obj/lastactivitytask.o .obj/bobfilecache.o .obj/statuspreset.o .obj/mucmanager.o .obj/mucconfigdlg.o .obj/mucaffiliationsmodel.o .obj/mucaffiliationsproxymodel.o .obj/mucaffiliationsview.o .obj/mucreasonseditor.o .obj/rosteritemexchangetask.o .obj/mood.o .obj/moodcatalog.o .obj/mooddlg.o .obj/activity.o .obj/activitycatalog.o .obj/activitydlg.o .obj/geolocation.o .obj/urlbookmark.o .obj/conferencebookmark.o .obj/bookmarkmanager.o .obj/pepmanager.o .obj/pubsubsubscription.o .obj/rc.o .obj/passdialog.o .obj/httpauthmanager.o .obj/ahcommand.o .obj/ahcommandserver.o .obj/ahcommanddlg.o .obj/ahcformdlg.o .obj/ahcexecutetask.o .obj/ahcservermanager.o .obj/serverlistquerier.o .obj/psioptions.o .obj/psioptionseditor.o .obj/voicecalldlg.o .obj/resourcemenu.o .obj/shortcutmanager.o .obj/bosskey.o .obj/psicontactlist.o .obj/psicon.o .obj/psiaccount.o .obj/accountlabel.o .obj/bookmarkmanagedlg.o .obj/vcardphotodlg.o .obj/coloropt.o .obj/sendbuttonmenu.o .obj/geolocationdlg.o .obj/rosteravatarframe.o .obj/tabcompletion.o .obj/psicapsregsitry.o .obj/alertmanager.o .obj/mcmdcompletion.o .obj/captchadlg.o .obj/filesharedlg.o .obj/multifiletransfermodel.o .obj/multifiletransferitem.o .obj/multifiletransferdelegate.o .obj/filesharingmanager.o .obj/filesharingdownloader.o .obj/filesharingitem.o .obj/filesharingnamproxy.o .obj/httputil.o .obj/filetransdlg.o .obj/groupchatdlg.o .obj/gcuserview.o .obj/mucjoindlg.o .obj/groupchattopicdlg.o .obj/contactlistview.o .obj/contactlistdragview.o .obj/hoverabletreeview.o .obj/contactlistmodel.o .obj/contactlistmodelselection.o .obj/contactlistdragmodel.o .obj/contactlistviewdelegate.o .obj/contactlistproxymodel.o .obj/psicontact.o .obj/psicontactmenu.o .obj/invitetogroupchatmenu.o .obj/groupmenu.o .obj/contactlistgroupmenu.o .obj/contactlistaccountmenu.o .obj/contactlistitem.o .obj/contactlistitemmenu.o .obj/contactupdatesmanager.o .obj/statusmenu.o .obj/globalstatusmenu.o .obj/accountstatusmenu.o .obj/psirosterwidget.o .obj/psifilteredcontactlistview.o .obj/abstracttreeitem.o .obj/abstracttreemodel.o .obj/psicontactlistview.o .obj/pgputil.o .obj/pgpkeydlg.o .obj/globaleventqueue.o .obj/dummystream.o .obj/networkaccessmanager.o .obj/bytearrayreply.o .obj/http_parser.o .obj/qhttpabstracts.o .obj/qhttpserverconnection.o .obj/qhttpserverrequest.o .obj/qhttpserverresponse.o .obj/qhttpserver.o .obj/filesharinghttpproxy.o .obj/webserver.o .obj/optionstab.o .obj/optionsdlgbase.o .obj/optionsdlg.o .obj/opt_application.o .obj/opt_roster.o .obj/opt_roster_main.o .obj/opt_roster_muc.o .obj/opt_chat.o .obj/opt_events.o .obj/opt_popups.o .obj/opt_status.o .obj/opt_statusgeneral.o .obj/opt_statusauto.o .obj/opt_appearance.o .obj/opt_iconset.o .obj/opt_input.o .obj/opt_messages.o .obj/opt_messages_common.o .obj/opt_theme.o .obj/opt_groupchat.o .obj/opt_sound.o .obj/opt_avcall.o .obj/opt_toolbars.o .obj/opt_advanced.o .obj/opt_shortcuts.o .obj/opt_statuspep.o .obj/opt_accounts.o .obj/opt_tree.o .obj/opt_plugins.o .obj/qite.o .obj/qiteaudio.o .obj/qiteprogress.o .obj/qiteaudiorecorder.o .obj/pluginmanager.o .obj/pluginhost.o .obj/dbus.o .obj/activeprofiles_dbus.o .obj/psidbusnotifier.o .obj/x11windowsystem.o .obj/chatview_webkit.o .obj/webview.o .obj/jsutil.o .obj/chatviewtheme.o .obj/chatviewthemeprovider.o .obj/chatviewthemeprovider_priv.o .obj/main.o .obj/multifiletransferdlg.o .obj/qrc_psi.o .obj/qrc_iconsets.o .obj/moc_discoinfoquerier.o .obj/moc_iris_discoinfoquerier.o .obj/moc_privacymanager.o .obj/moc_psiprivacymanager.o .obj/moc_privacydlg.o .obj/moc_privacyruledlg.o .obj/moc_tabdlg.o .obj/moc_tabmanager.o .obj/moc_tabbablewidget.o .obj/moc_CertificateDisplayDialog.o .obj/moc_CertificateErrorDialog.o .obj/moc_contactmanagerdlg.o .obj/moc_contactmanagermodel.o .obj/moc_contactmanagerview.o .obj/moc_idle.o .obj/moc_systemwatch.o .obj/moc_systemwatch_unix.o .obj/moc_globalshortcuttrigger.o .obj/moc_priorityvalidator.o .obj/moc_iconset.o .obj/moc_optionstree.o .obj/moc_varianttree.o .obj/moc_optionstreemodel.o .obj/moc_advwidget.o .obj/moc_grepshortcutkeydialog.o .obj/moc_psimedia.o .obj/moc_psimedia_p.o .obj/moc_jinglertptasks.o .obj/moc_jinglertp.o .obj/moc_avcall.o .obj/moc_calldlg.o .obj/moc_mediadevicewatcher.o .obj/moc_tunecontroller.o .obj/moc_pollingtunecontroller.o .obj/moc_filetunecontroller.o .obj/moc_tunecontrollermanager.o .obj/moc_psifileplugin.o .obj/moc_mpristunecontroller.o .obj/moc_busywidget.o .obj/moc_fancylabel.o .obj/moc_iconselect.o .obj/moc_iconsetselect.o .obj/moc_iconsetdisplay.o .obj/moc_iconwidget.o .obj/moc_iconbutton.o .obj/moc_psitextview.o .obj/moc_iconaction.o .obj/moc_fancypopup.o .obj/moc_urllabel.o .obj/moc_urlobject.o .obj/moc_updatingcombobox.o .obj/moc_iconlabel.o .obj/moc_icontoolbutton.o .obj/moc_fancypopuplist.o .obj/moc_psitiplabel.o .obj/moc_psitabwidget.o .obj/moc_psitabbar.o .obj/moc_actionlineedit.o .obj/moc_tabbar.o .obj/moc_typeaheadfind.o .obj/moc_pixmapratiolabel.o .obj/moc_showtextdlg.o .obj/moc_activeprofiles.o .obj/moc_profiledlg.o .obj/moc_aboutdlg.o .obj/moc_fileutil.o .obj/moc_psiaccount.o .obj/moc_psicon.o .obj/moc_accountscombobox.o .obj/moc_psievent.o .obj/moc_xmlconsole.o .obj/moc_psiiconset.o .obj/moc_psithememanager.o .obj/moc_psithememodel.o .obj/moc_psithemeprovider.o .obj/moc_pgptransaction.o .obj/moc_mainwin.o .obj/moc_mainwin_p.o .obj/moc_psitrayicon.o .obj/moc_proxy.o .obj/moc_miniclient.o .obj/moc_accountmanagedlg.o .obj/moc_accountadddlg.o .obj/moc_accountregdlg.o .obj/moc_accountmodifydlg.o .obj/moc_changepwdlg.o .obj/moc_htmltextcontroller.o .obj/moc_msgmle.o .obj/moc_statusdlg.o .obj/moc_statuscombobox.o .obj/moc_eventdlg.o .obj/moc_activecontactsmenu.o .obj/moc_chatdlg.o .obj/moc_psichatdlg.o .obj/moc_chatsplitter.o .obj/moc_chateditproxy.o .obj/moc_adduserdlg.o .obj/moc_mcmdmanager.o .obj/moc_infodlg.o .obj/moc_translationmanager.o .obj/moc_eventdb.o .obj/moc_edbflatfile.o .obj/moc_edbsqlite.o .obj/moc_historydlg.o .obj/moc_historyimp.o .obj/moc_historycontactlistmodel.o .obj/moc_searchdlg.o .obj/moc_registrationdlg.o .obj/moc_psitoolbar.o .obj/moc_passphrasedlg.o .obj/moc_vcardfactory.o .obj/moc_tasklist.o .obj/moc_discodlg.o .obj/moc_alerticon.o .obj/moc_alertable.o .obj/moc_psipopup.o .obj/moc_psiapplication.o .obj/moc_filecache.o .obj/moc_avatars.o .obj/moc_actionlist.o .obj/moc_xdata_widget.o .obj/moc_bobfilecache.o .obj/moc_mucmanager.o .obj/moc_mucconfigdlg.o .obj/moc_mucaffiliationsmodel.o .obj/moc_mucaffiliationsproxymodel.o .obj/moc_mucaffiliationsview.o .obj/moc_mucreasonseditor.o .obj/moc_rosteritemexchangetask.o .obj/moc_mooddlg.o .obj/moc_activitydlg.o .obj/moc_bookmarkmanager.o .obj/moc_pepmanager.o .obj/moc_passdialog.o .obj/moc_httpauthmanager.o .obj/moc_ahcommanddlg.o .obj/moc_ahcformdlg.o .obj/moc_serverlistquerier.o .obj/moc_psioptionseditor.o .obj/moc_psioptions.o .obj/moc_voicecaller.o .obj/moc_voicecalldlg.o .obj/moc_resourcemenu.o .obj/moc_bosskey.o .obj/moc_psicontactlist.o .obj/moc_accountlabel.o .obj/moc_bookmarkmanagedlg.o .obj/moc_vcardphotodlg.o .obj/moc_psicli.o .obj/moc_coloropt.o .obj/moc_sendbuttonmenu.o .obj/moc_geolocationdlg.o .obj/moc_rosteravatarframe.o .obj/moc_psicapsregsitry.o .obj/moc_tabcompletion.o .obj/moc_alertmanager.o .obj/moc_captchadlg.o .obj/moc_filesharedlg.o .obj/moc_multifiletransfermodel.o .obj/moc_multifiletransferitem.o .obj/moc_multifiletransferdelegate.o .obj/moc_filesharingmanager.o .obj/moc_filesharingdownloader.o .obj/moc_filesharingitem.o .obj/moc_filesharingnamproxy.o .obj/moc_filetransdlg.o .obj/moc_groupchatdlg.o .obj/moc_gcuserview.o .obj/moc_mucjoindlg.o .obj/moc_groupchattopicdlg.o .obj/moc_contactlistview.o .obj/moc_contactlistdragview.o .obj/moc_hoverabletreeview.o .obj/moc_contactlistmodel.o .obj/moc_contactlistmodel_p.o .obj/moc_contactlistmodelselection.o .obj/moc_contactlistdragmodel.o .obj/moc_contactlistviewdelegate.o .obj/moc_contactlistviewdelegate_p.o .obj/moc_contactlistproxymodel.o .obj/moc_psicontact.o .obj/moc_psicontactmenu.o .obj/moc_psicontactmenu_p.o .obj/moc_groupmenu.o .obj/moc_invitetogroupchatmenu.o .obj/moc_contactlistgroupmenu.o .obj/moc_contactlistgroupmenu_p.o .obj/moc_contactlistaccountmenu.o .obj/moc_contactlistitemmenu.o .obj/moc_contactupdatesmanager.o .obj/moc_statusmenu.o .obj/moc_globalstatusmenu.o .obj/moc_accountstatusmenu.o .obj/moc_psirosterwidget.o .obj/moc_psifilteredcontactlistview.o .obj/moc_abstracttreemodel.o .obj/moc_psicontactlistview.o .obj/moc_pgputil.o .obj/moc_pgpkeydlg.o .obj/moc_globaleventqueue.o .obj/moc_networkaccessmanager.o .obj/moc_bytearrayreply.o .obj/moc_qhttpabstracts.o .obj/moc_qhttpserverconnection.o .obj/moc_qhttpserverrequest.o .obj/moc_qhttpserverresponse.o .obj/moc_qhttpserver.o .obj/moc_filesharinghttpproxy.o .obj/moc_webserver.o .obj/moc_optionstab.o .obj/moc_optionsdlgbase.o .obj/moc_optionsdlg.o .obj/moc_opt_application.o .obj/moc_opt_roster.o .obj/moc_opt_roster_main.o .obj/moc_opt_roster_muc.o .obj/moc_opt_chat.o .obj/moc_opt_events.o .obj/moc_opt_popups.o .obj/moc_opt_status.o .obj/moc_opt_statusgeneral.o .obj/moc_opt_statusauto.o .obj/moc_opt_appearance.o .obj/moc_opt_iconset.o .obj/moc_opt_input.o .obj/moc_opt_messages.o .obj/moc_opt_messages_common.o .obj/moc_opt_theme.o .obj/moc_opt_groupchat.o .obj/moc_opt_sound.o .obj/moc_opt_avcall.o .obj/moc_opt_toolbars.o .obj/moc_opt_advanced.o .obj/moc_opt_shortcuts.o .obj/moc_opt_statuspep.o .obj/moc_opt_accounts.o .obj/moc_opt_tree.o .obj/moc_opt_plugins.o .obj/moc_qite.o .obj/moc_qiteaudio.o .obj/moc_qiteprogress.o .obj/moc_qiteaudiorecorder.o .obj/moc_pluginmanager.o .obj/moc_pluginhost.o .obj/moc_chatview_webkit.o .obj/moc_webview.o .obj/moc_chatviewtheme.o .obj/moc_chatviewtheme_p.o .obj/moc_chatviewthemeprovider.o .obj/moc_chatviewthemeprovider_priv.o .obj/moc_main.o .obj/moc_multifiletransferdlg.o   -lxcb -L/home/albert/Downloads/git/psi/iris/lib -liris -lirisnet -L/usr/lib64 -lidn -lz -lhunspell-1.7 -lqca-qt5 -lpthread /usr/lib64/libQt5WebEngineWidgets.so /usr/lib64/libQt5PrintSupport.so /usr/lib64/libQt5Widgets.so /usr/lib64/libQt5Multimedia.so /usr/lib64/libQt5X11Extras.so /usr/lib64/libQt5WebEngineCore.so /usr/lib64/libQt5Quick.so /usr/lib64/libQt5Gui.so /usr/lib64/libQt5Xml.so /usr/lib64/libQt5WebChannel.so /usr/lib64/libQt5Qml.so /usr/lib64/libQt5Network.so /usr/lib64/libQt5Sql.so /usr/lib64/libQt5Concurrent.so /usr/lib64/libQt5DBus.so /usr/lib64/libQt5Positioning.so /usr/lib64/libQt5Core.so /usr/lib64/libGL.so -lXext -lX11 -lm
/usr/bin/ld: .obj/common.o: in function `QtPrivate::QFunctorSlotObject<saveXMPPPasswordToKeyring(QString const&, QString const&, QObject*)::{lambda(QKeychain::Job*)#1}, 1, QtPrivate::List<QKeychain::Job*>, void>::impl(int, QtPrivate::QSlotObjectBase*, QObject*, void**, bool*)':
common.cpp:(.text+0x15): undefined reference to `QKeychain::Job::error() const'
/usr/bin/ld: .obj/common.o: in function `saveXMPPPasswordToKeyring(QString const&, QString const&, QObject*)':
common.cpp:(.text+0x728): undefined reference to `QKeychain::WritePasswordJob::WritePasswordJob(QString const&, QObject*)'
/usr/bin/ld: common.cpp:(.text+0x73b): undefined reference to `QKeychain::WritePasswordJob::setTextData(QString const&)'
/usr/bin/ld: common.cpp:(.text+0x74b): undefined reference to `QKeychain::Job::setKey(QString const&)'
/usr/bin/ld: common.cpp:(.text+0x758): undefined reference to `QKeychain::Job::setAutoDelete(bool)'
/usr/bin/ld: common.cpp:(.text+0x75f): undefined reference to `QKeychain::Job::finished(QKeychain::Job*)'
/usr/bin/ld: common.cpp:(.text+0x7a4): undefined reference to `QKeychain::Job::staticMetaObject'
/usr/bin/ld: common.cpp:(.text+0x7ce): undefined reference to `QKeychain::Job::start()'
/usr/bin/ld: .obj/accountmodifydlg.o: in function `QtPrivate::QFunctorSlotObject<AccountModifyDlg::save()::{lambda(QKeychain::Job*)#1}, 1, QtPrivate::List<QKeychain::Job*>, void>::impl(int, QtPrivate::QSlotObjectBase*, QObject*, void**, bool*)':
accountmodifydlg.cpp:(.text+0x19): undefined reference to `QKeychain::Job::error() const'
/usr/bin/ld: .obj/accountmodifydlg.o: in function `QtPrivate::QFunctorSlotObject<AccountModifyDlg::init()::{lambda(QKeychain::Job*)#1}, 1, QtPrivate::List<QKeychain::Job*>, void>::impl(int, QtPrivate::QSlotObjectBase*, QObject*, void**, bool*)':
accountmodifydlg.cpp:(.text+0x90): undefined reference to `QKeychain::Job::error() const'
/usr/bin/ld: accountmodifydlg.cpp:(.text+0xdf): undefined reference to `QKeychain::ReadPasswordJob::textData() const'
/usr/bin/ld: .obj/accountmodifydlg.o: in function `AccountModifyDlg::init()':
accountmodifydlg.cpp:(.text+0x1714): undefined reference to `QKeychain::ReadPasswordJob::ReadPasswordJob(QString const&, QObject*)'
/usr/bin/ld: accountmodifydlg.cpp:(.text+0x1729): undefined reference to `QKeychain::Job::setKey(QString const&)'
/usr/bin/ld: accountmodifydlg.cpp:(.text+0x1736): undefined reference to `QKeychain::Job::setAutoDelete(bool)'
/usr/bin/ld: accountmodifydlg.cpp:(.text+0x173d): undefined reference to `QKeychain::Job::finished(QKeychain::Job*)'
/usr/bin/ld: accountmodifydlg.cpp:(.text+0x178c): undefined reference to `QKeychain::Job::staticMetaObject'
/usr/bin/ld: accountmodifydlg.cpp:(.text+0x17a9): undefined reference to `QKeychain::Job::start()'
/usr/bin/ld: .obj/accountmodifydlg.o: in function `AccountModifyDlg::save()':
accountmodifydlg.cpp:(.text+0x4d4c): undefined reference to `QKeychain::WritePasswordJob::WritePasswordJob(QString const&, QObject*)'
/usr/bin/ld: accountmodifydlg.cpp:(.text+0x4d63): undefined reference to `QKeychain::WritePasswordJob::setTextData(QString const&)'
/usr/bin/ld: accountmodifydlg.cpp:(.text+0x4d72): undefined reference to `QKeychain::Job::setKey(QString const&)'
/usr/bin/ld: accountmodifydlg.cpp:(.text+0x4d7f): undefined reference to `QKeychain::Job::setAutoDelete(bool)'
/usr/bin/ld: accountmodifydlg.cpp:(.text+0x4d86): undefined reference to `QKeychain::Job::finished(QKeychain::Job*)'
/usr/bin/ld: accountmodifydlg.cpp:(.text+0x4de1): undefined reference to `QKeychain::Job::staticMetaObject'
/usr/bin/ld: accountmodifydlg.cpp:(.text+0x4dfe): undefined reference to `QKeychain::Job::start()'
/usr/bin/ld: accountmodifydlg.cpp:(.text+0x5165): undefined reference to `QKeychain::DeletePasswordJob::DeletePasswordJob(QString const&, QObject*)'
/usr/bin/ld: .obj/psiaccount.o: in function `PsiAccount::cs_needAuthParams(bool, bool, bool)':
psiaccount.cpp:(.text+0x150d): undefined reference to `QKeychain::ReadPasswordJob::ReadPasswordJob(QString const&, QObject*)'
/usr/bin/ld: psiaccount.cpp:(.text+0x152e): undefined reference to `QKeychain::Job::setKey(QString const&)'
/usr/bin/ld: psiaccount.cpp:(.text+0x153b): undefined reference to `QKeychain::Job::setAutoDelete(bool)'
/usr/bin/ld: psiaccount.cpp:(.text+0x1542): undefined reference to `QKeychain::Job::finished(QKeychain::Job*)'
/usr/bin/ld: psiaccount.cpp:(.text+0x158b): undefined reference to `QKeychain::Job::staticMetaObject'
/usr/bin/ld: .obj/psiaccount.o: in function `QtPrivate::QFunctorSlotObject<PsiAccount::cs_needAuthParams(bool, bool, bool)::{lambda(QKeychain::Job*)#1}, 1, QtPrivate::List<QKeychain::Job*>, void>::impl(int, QtPrivate::QSlotObjectBase*, QObject*, void**, bool*)':
psiaccount.cpp:(.text+0xeb8a): undefined reference to `QKeychain::Job::error() const'
/usr/bin/ld: psiaccount.cpp:(.text+0xebe2): undefined reference to `QKeychain::ReadPasswordJob::textData() const'
/usr/bin/ld: psiaccount.cpp:(.text+0xecb1): undefined reference to `QKeychain::Job::error() const'
/usr/bin/ld: .obj/psiaccount.o: in function `PsiAccount::cs_needAuthParams(bool, bool, bool)':
psiaccount.cpp:(.text+0x15b4): undefined reference to `QKeychain::Job::start()'
collect2: error: ld returned 1 exit status
make[1]: *** [Makefile:1887: /home/albert/Downloads/git/psi/psi] Error 1
make[1]: Leaving directory '/home/albert/Downloads/git/psi/src'
make: *** [Makefile:73: sub-src-make_first] Error 2
```
</details>

**Comment by Ri0n** [comment]
It's a QtKeychain bug. They have fixed this problem in master but didn't release it.
As a workaround you can build Psi with cmake.
I'll try to add some early detection of broken qtkeychain.
Some distros like Suse already include patches.

**Comment by algo99** [comment]
Thank you for the fast reply !
A bit dirty workaround is to append `/usr/lib64/libkeychain.so` to the `LIB` variable in `src/Makefile`

**Comment by Ri0n** [comment]
ah. I just checked qtkeychain repo. They released it! The fix is available in 0.10.0

For manual fix see this https://github.com/frankosterfeld/qtkeychain/commit/4a9e58b0eac4af17dfca4ccd0796a44244700991

**Comment by algo99** [comment]
Ok, thank you.

**Comment by Ri0n** [comment]
fixed in 3e09a32a097da798987986e15f6fd32944858253
